zcash_keys/src/keys.rs

```diff
@@ -492,6 +495,29 @@ pub enum AddressGenerationError {
     /// A Unified address cannot be generated without at least one shielded receiver being
     /// included.
     ShieldedReceiverRequired,
+    InvalidShieldedKey(ShieldedProtocol),
+    InvalidTransparentKey,
+}
+
+impl From<UnifiedError> for AddressGenerationError {
+    fn from(e: UnifiedError) -> Self {
+        match e {
+            UnifiedError::InvalidShieldedKey(p) => AddressGenerationError::InvalidShieldedKey(p),
+            // We have to drop the transparent key error because that enum type in hdwallet does not implement Clone.
+            UnifiedError::InvalidTransparentKey(_) => AddressGenerationError::InvalidTransparentKey,
+        }
+    }
+}
+
+impl From<hdwallet::error::Error> for AddressGenerationError {
+    fn from(e: hdwallet::error::Error) -> Self {
+        match e {
+            hdwallet::error::Error::KeyIndexOutOfRange => todo!(),
+            hdwallet::error::Error::ChainPath(_) => todo!(),
+            hdwallet::error::Error::Secp(_) => todo!(),
+            hdwallet::error::Error::Rng(_) => todo!(),
+        }
```

Comment:
Instead of this, just wrap the entire `hdwallet::error::Error` in a new `AddressGenerationError` variant.

Reply:
I can't, because the hdwallet Error enum doesn't derive the same traits as `AddressGenerationError`. I'd love to though. Time for an official fork of hdwallet?

Reply:
What traits would be restricted to? We may not be using all of the traits we derive.

Reply:
`hdwallet` derives `Error` and `Debug`. `AddressGenerationError` adds `Clone`.

Reply:
Removing the `Clone` trait didn't break anything within librustzcash itself. So as it sounds like you were suggesting that approach, I'm taking it.